lib/ast-utils.js

```diff
@@ -290,8 +290,13 @@ const lineIndexCache = new WeakMap();
 function getLineIndices(sourceCode) {
 
     if (!lineIndexCache.has(sourceCode)) {
-        const lineIndices = (sourceCode.text.match(/[^\r\n\u2028\u2029]*(\r\n|\r|\n|\u2028|\u2029)/g) || [])
-            .reduce((indices, line) => indices.concat(indices[indices.length - 1] + line.length), [0]);
+        const lineIndices = [0];
```

Comment:
Can you add a comment here explaining why it's done this way? I can see some well-meaning dev thinking they can simplify this and reverting accidentally.